components/edit-collective/sections/UserSecurity.js

```diff
@@ -194,6 +212,143 @@ class UserTwoFactorAuth extends React.Component {
+    const { password, passwordError, passwordLoading, passwordKey, currentPassword } = this.state;
+
+    return (
+      <Fragment>
+        <H3 fontSize="18px" fontWeight="700" mb={2}>
+          <FormattedMessage id="Password" defaultMessage="Password" />
+        </H3>
+        {passwordError && (
+          <MessageBox type="error" withIcon my={2} data-cy="password-error">
+            {passwordError}
+          </MessageBox>
+        )}
+        <Container mb="4">
+          <P py={2} mb={2}>
+            {LoggedInUser.hasPassword ? (
+              <FormattedMessage
+                id="Password.Change.Info"
+                defaultMessage="You already have a password set, you can change it using the following form."
+              />
+            ) : (
+              <FormattedMessage
+                id="Password.Set.Info"
+                defaultMessage="Setting a password is optional but this can be useful for regular users."
+              />
+            )}
+          </P>
+
+          {LoggedInUser.hasPassword && (
+            <StyledInputField
+              label={<FormattedMessage defaultMessage="Current Password" />}
+              labelFontWeight="bold"
+              htmlFor="current-password"
+              mb={2}
+              width="100%"
+            >
+              <StyledInput
+                key={`current-password-${passwordKey}`}
+                fontSize="14px"
+                id="current-password"
+                autoComplete="current-password"
+                name="current-password"
+                type="password"
+                required
+                onChange={e => {
+                  this.setState({ passwordError: null, currentPassword: e.target.value });
+                }}
+              />
+            </StyledInputField>
+          )}
+
+          <StyledInputField
+            label={<FormattedMessage defaultMessage="New Password" />}
+            labelFontWeight="bold"
+            htmlFor="new-password"
+            mt={2}
+            mb={2}
+            width="100%"
+          >
+            <StyledInput
+              key={`current-password-${passwordKey}`}
+              fontSize="14px"
+              id="new-password"
+              autoComplete="new-password"
+              type="password"
+              required
+              onChange={e => {
+                this.setState({ passwordError: null, password: e.target.value });
+              }}
+            />
+          </StyledInputField>
+
+          <PasswordStrengthBar
+            style={{ visibility: password ? 'visible' : 'hidden' }}
+            password={this.state.password}
+            onChangeScore={passwordScore => {
+              this.setState({ passwordScore });
+            }}
+          />
+
+          <StyledButton
+            my={2}
+            minWidth={140}
+            disabled={passwordLoading || !password || (LoggedInUser.hasPassword && !currentPassword)}
+            onClick={this.setPassword}
```

Comment:
Loading state makes it clearer that the form is being submitted and not just blocked

```suggestion
            loading={passwordLoading}
            disabled={!password || (LoggedInUser.hasPassword && !currentPassword)}
            onClick={this.setPassword}
```